p2p/swarmhandlers.go

```diff
@@ -402,6 +419,7 @@ func (s *swarmImpl) onRemoteClientHandshakeMessage(msg net.IncomingMessage) {
 	s.localNode.Debug("Routing message from %v to handshake handler ", sender.Pretty())
 	// Let the demuxer route the message to its registered protocol handler
 	s.demuxer.RouteIncomingMessage(NewIncomingMessage(sender, data.Protocol, msg.Message))
+	s.peerByConMapMutex.Unlock()
```

Comment:
Why do you need to keep the mutex locked until this point? Why not unlocking it before the 'if'?